docs: merge deployment pages in getting started (#7765)
nuxt/framework: docs/content/1.getting-started/10.deployment.md

# Deployment

> A Nuxt application can be deployed on a Node or Deno server, pre-rendered to be hosted in static environments, or deployed to serverless and edge providers.

Reviewer: What about putting the specific providers at top, as well as links to Node + SSG? Those two sections are quite lengthy and would be good to give people an overview before diving into the details.

Author: Agree, I think that a link on top is a good first step. The list of providers could also be displayed (in a future PR) as a cards grid, which would allow providers to fill a short text with their key points.

- `NITRO_PORT` or `PORT` (defaults to `3000`)
- `NITRO_HOST` or `HOST` (defaults to `'0.0.0.0'`)
- `NITRO_SSL_CERT` and `NITRO_SSL_KEY` - if both are present, this will launch the server in HTTPS mode. In the vast majority of cases, this should not be used other than for testing, and the Nitro server should be run behind a reverse proxy like nginx or Cloudflare which terminates SSL.

#### Using PM2

To use `pm2`, use an `ecosystem.config.js`:

```js [ecosystem.config.js]
module.exports = {
  apps: [
    {
      name: 'NuxtAppName',
      exec_mode: 'cluster',
      instances: 'max',
      script: './.output/server/index.mjs'
    }
  ]
}
```

#### Using Cluster Mode

You can use `NITRO_PRESET=node_cluster` in order to leverage multi-process performance using Node.js [cluster](https://nodejs.org/dist/latest/docs/api/cluster.html) module.

By default, the workload gets distributed to the workers with the round robin strategy.

### Learn More

:ReadMore{link="https://nitro.unjs.io/deploy/node" title="the Nitro documentation for node-server preset"}

## Static Hosting

There are two ways to deploy a Nuxt application to any static hosting services:

> - Static site generation (SSG) prerenders every route of your application at build time. For every page, Nuxt uses a crawler to generate a corresponding HTML and payload files.
> - Using `ssr: false` to produce a pure client-side output.
> 
> ### Prerendering

Reviewer 2: ```suggestion
- Static site generation (SSG) pre-renders every route of your application at build time. For every page, Nuxt uses a crawler to generate a corresponding HTML and payload files.
- Using `ssr: false` to produce a pure client-side output.

### Pre-rendering
```

Use the [`nuxi generate` command](/api/commands/generate) to build your application. The HTML files will be generated in the `.output/public` directory.

```bash
npx nuxi generate
```

### Client-side Only Rendering

> If you don't want to prerender your routes, another way of using static hosting is to set the `ssr` property to `false` in the `nuxt.config` file. The `nuxi generate` command will then output an `index.html` entrypoint like a classic client-side Vue.js application.

Reviewer 2: ```suggestion
If you don't want to pre-render your routes, another way of using static hosting is to set the `ssr` property to `false` in the `nuxt.config` file. The `nuxi generate` command will then output an `index.html` entrypoint like a classic client-side Vue.js application.
```

> If you don't want to prerender your routes, another way of using static hosting is to set the `ssr` property to `false` in the `nuxt.config` file. The `nuxi generate` command will then output an `index.html` entrypoint like a classic client-side Vue.js application.

Reviewer 3: Should we say .output/public/index.html instead?

What about 200.html?

```ts [nuxt.config.ts|js]
defineNuxtConfig({
  ssr: false
})
```

### Advanced

> You can manually specify routes that [Nitro](/guide/concepts/server-engine) will fetch and prerender during the build.

Reviewer 2: ```suggestion
You can manually specify routes that [Nitro](/guide/concepts/server-engine) will fetch and pre-render during the build.
```

> You can manually specify routes that [Nitro](/guide/concepts/server-engine) will fetch and prerender during the build.

Reviewer 3: I would add this section before the client side only personally

```ts [nuxt.config.ts|js]
defineNuxtConfig({
  nitro: {
    prerender: {
      routes: ['/user/1', '/user/2']
    }
  }
})
```

## Presets

In addition to Node.js servers and static hosting services, a Nuxt 3 project can be deployed with several well-tested presets and minimal amount of configuration.

> You can use the [Nuxt config](/guide/directory-structure/nuxt.config) to explicitly set the preset to use:

Reviewer 2: ```suggestion
You can explicitly set the desired preset in the [`nuxt.config`](/guide/directory-structure/nuxt.config) file:
```

```js [nuxt.config.js|ts]
export default {
  nitro: {
    preset: 'node-server'
  }
}
```

> Or directly use the `NITRO_PRESET` environment variable when running `nuxt build`:

Reviewer 2: ```suggestion
... or use the `NITRO_PRESET` environment variable when running `nuxt build`:
```

# Deployment

> A Nuxt application can be deployed on a Node or Deno server, pre-rendered to be hosted in static environments, or deployed to serverless and edge providers.

Reviewer 4: Deno is still not supported

```suggestion
A Nuxt application can be deployed on a Node.js server, pre-rendered to be hosted in static environments, or deployed to serverless and edge providers in CDN.
```

> A Nuxt application can be deployed on a Node.js server, pre-rendered to be hosted in static environments, or deployed to serverless and edge providers in CDN.

Reviewer: ```suggestion
A Nuxt application can be deployed on a Node.js server, pre-rendered for static hosting, or deployed to serverless or edge (CDN) environments.
```
